enhancements/cluster-logging/loki-gateway-configuration.md

```diff
@@ -0,0 +1,489 @@
+
+```
+kubectl create secret generic lokiStack-gateway-tenant-a --from-literal='clientID=test' --from-literal='clientSecret=ZXhhbXBsZS1hcHAtc2VjcmV0' --from-literal='issuerCAPath=/tmp/certs/ca.pem'
+```
+
+### Creating runtime configMap
+
+If the lokiStack CR provides oidc configuration then build a `gateway-rbac-runtime.yaml` and `gateway-tenants-runtime.yaml` file in the following way:
+
+#### gateway-tenants-runtime.yaml
+
+```yaml
+---
+overrides:
+  tenants:
+  {{- range $tenant, $spec := .Stack.Template.GatewayConfiguration.Tenants }}
+  - name: {{ $spec.Name }}
+    id: {{ $spec.ID }}
+  {{- end -}}
+```
+
+#### gateway-rbac-runtime.yaml
+
+```yaml
+---
+overrides:
+  roleBindings:
+  {{- range $roleBinding, $spec := .Stack.Template.GatewayConfiguration.Rbac.RoleBindings }}
+  - {{ $roleBinding }}: {{ $spec.Name }}
+  {{- end -}}
+  roles:
+  {{- range $role, $spec := .Stack.Template.GatewayConfiguration.Rbac.Roles }}
+  - {{ $role }}: {{ $spec.Name }}
+  {{- end -}}
+```
+
+### Test Plan
+
+#### Unit Testing
+
+The [loki-operator](https://github.com/viaq/loki-operator) includes a framework based on [counterfeiter](github.com/maxbrunsfeld/counterfeiter) to create simple and useful fake client to test configuration in unit tests.
+
+Testing of the reconciliation of a `LokiStack` custom resource will be based upon the same technique where the custom resource describes the inputs and the generated manifests the outputs.
+
+#### Functional Testing
+
+The loki-operator uses [testify](github.com/stretchr/testify) to do functional testing.
+
+- Verify that the user is able to configure the modes correctly.
+- Verify that the configMap is correctly getting generated with both tenants and rbac information.
+- Verify that the secrets are correctly mapped to it's corresponding OIDCSpec.
+- Verify that the runtime rbac and tenant values take precedence over the default values.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback by switching the internal Loki service to be managed by the [loki-operator](https://github.com/viaq/loki-operator).
+
+#### Tech Preview -> GA
+
+- More testing
+- Sufficient time for feedback
+- Available by default
+- Conduct load testing per t-shirt size supported.
+
+### Upgrade / Downgrade Strategy
+
+Not applicable here.
+
+### Version Skew Strategy
+
+Given we are adding a new API there should be minimal concern for upgrading.
+
+## Implementation History
+
+| Release | Description              |
+| ---     | ---                      |
+| TBD     | **TP** - Initial release |
+| TBD     | **GA** - Initial release |
+
+## Drawbacks
+
+The drawback to not implementing this feature is we are unable to integrate Loki access to our org's OIDC provider and the current multi-tenant featurs of OpenShift Logging
```

Comment:
```suggestion
The drawback to not implementing this feature is we are unable to integrate Loki access to our org's OIDC provider and the current multi-tenant features of OpenShift Logging
```

Why? 🤗

Reply:
@bwplotka The reason is to enable namespace-based tenancy as per OpenShift maintains this world view on resources. The developer persona has access only to namespaces (most of the times only application workload ns) granted by RBAC. The administrator persona has access to all namespaces including audit logs.

@sasagarw I suggest to add a couple of lines explaining this :point_up: